### ISSUE TITLE
Missing assignments from assembly in decompiled output 2!

### ISSUE THREAD

**Issue by Wall-AF** [NationalSecurityAgency/ghidra#4620]
When a parameter to an indirectly called function is a structure and has had its components assigned from elsewhere, those assignments don't appear in the decompilation.

I'm unsure if this is due to the fact that in the assembly the register used to temporarily hold the value being copied is an extended register `EAX` and my code is using the `x86-16.?spec` configuration files. If so, do I need to add these extended registers to any config files and how?

Here's an example:
[dragon_Write_11e0_1a94.zip](https://github.com/NationalSecurityAgency/ghidra/files/9583379/dragon_Write_11e0_1a94.zip)

_Originally posted by @Wall-AF in https://github.com/NationalSecurityAgency/ghidra/issues/4529#issuecomment-1249221954_

**Comment by Wall-AF** [comment]
Don't know if this is related, but when decompiling indirect calls (using mnemonics `CALL` & `CALLF`) the stack depth has to be manually set!

**Comment by Wall-AF** [comment]
Here's another example: [dragon_readPicStates.zip](https://github.com/NationalSecurityAgency/ghidra/files/9939666/dragon_readPicStates.zip)
Please ask if you need anything more.

**Comment by Wall-AF** [comment]
[Here](https://github.com/NationalSecurityAgency/ghidra/files/9994860/dragon_FUN_1048_0f7d.zip) is a further example in which the assignment omission breaks the semantics of the function (see assembly at `1048:1093`).

**Comment by Wall-AF** [comment]
@ryanmkurtz would you help me to understand why/how this one-liner fixes this issue please.  I'm looking for a way to fix references from `far` pointers.

**Comment by ryanmkurtz** [comment]
Unfortunately I have no clue.  @caheckman will have to explain it.

**Comment by Wall-AF** [comment]
@caheckman @ryanmkurtz Please can this be re-opened as I've found yet another example with several missing assignments and computations that make the decompilation completely inaccurate in [this example](https://github.com/NationalSecurityAgency/ghidra/files/10478749/dragon_FUN_1110_06d0.zip) and shown highlighted below:
![image](https://user-images.githubusercontent.com/67698927/214017970-713c14de-0f3e-46c3-b404-5ff64931bc26.png)

**Comment by ryanmkurtz** [comment]
@Wall-AF, can you please open a new ticket for this problem?  

In the future, if a problem is presented in a ticket and we close the ticket with a fix to that problem, we'd like new problems in new tickets, even if they are seemingly related.  A couple reasons for this are:

1. Sometimes the new problem is actually unrelated, which isn't clear until we triage it.
2. Sometimes we will have cut a release before we can get to the newly presented problem and then the ticket's milestone gets out of whack.

Thanks!
